test/dialing/direct.spec.js

```diff
@@ -176,14 +177,36 @@ describe('Dialing (direct, WebSockets)', () => {
       .and.to.have.property('code', ErrorCodes.ERR_TIMEOUT)
   })
 
+  it('should sort addresses on dial', async () => {
+    sinon.spy(dialerUtils, 'sortPublicAddressesFirst')
```

Comment:
It would be good to actually verify the dial order here, we dont check anything after `connectToPeer`. Just pass a stub to `addressSorter` and return a different order than addressBook returns. We should be able to spy on the transportManager and ensure the addresses are executed in the same order `addressSorter` returns.